src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,624 @@
+
+        if (zeroDay < 0) {
+            throw new DateTimeException ("Negative epoch invalid: " + epochDay);
+        }
+
+        long yearEst = (400 * zeroDay) / DAYS_PER_CYCLE;
+        long doyEst = zeroDay - (365 * yearEst + yearEst / 4 - yearEst / 100 + yearEst / 400);
+        int month = 1 + (int) ((doyEst - 1 ) / 28);
+        int dom = 1 + (int) ((doyEst - 1) % 28);
+
+        if (month == 14) {
+            month = 13;
+            dom += 28;
+        }
+
+        // check year now we are certain it is correct
+        int year = ChronoField.YEAR.checkValidIntValue (yearEst);
+
+        return new InternationalFixedDate (year, month, dom);
+    }
+
+    private static InternationalFixedDate resolvePreviousValid (final int prolepticYear, final int month, final int day) {
+        int monthR = Math.min (month, InternationalFixedChronology.MONTHS_IN_YEAR);
+        int dayR = Math.min (day, InternationalFixedChronology.DAYS_IN_MONTH + (month == 6 || month == 13 ? 1 : 0));
+
+        return InternationalFixedDate.of (prolepticYear, monthR, dayR);
+    }
+
+    //-----------------------------------------------------------------------
+
+    /**
+     * Get the count of leap years since International fixed year 0.
+     * <p>
+     * This number is negative if the year is prior to International fixed year 0.
+     *
+     * @param prolepticYear The year.
+     * @return The number of leap years since International fixed year 0.
+     */
+    private static long getLeapYearsBefore (final long prolepticYear) {
+        return (prolepticYear / 4) - (prolepticYear / 100) + (prolepticYear / 400);
+    }
+
+    /**
+     * Validates the object.
+     *
+     * @return the resolved date, not null
+     */
+    private Object readResolve () {
+        return InternationalFixedDate.of (prolepticYear, month, day);
+    }
+
+    //-----------------------------------------------------------------------
+    @Override
+    int getProlepticYear () {
+        return prolepticYear;
+    }
+
+    @Override
+    int getMonth () {
+        return month;
+    }
+
+    @Override
+    int getDayOfMonth () {
+        return day;
+    }
+
+    @Override
+    int getDayOfYear () {
+        return (month - 1) * InternationalFixedChronology.DAYS_IN_MONTH + getDayOfMonth ();
+    }
+
+    @Override
+    AbstractDate withDayOfYear (final int value) {
+        return plusDays (value - getDayOfYear ());
+    }
+
+    @Override
+    int lengthOfYearInMonths () {
+        return InternationalFixedChronology.MONTHS_IN_YEAR;
+    }
+
+    @Override
+    ValueRange rangeAlignedWeekOfMonth () {
+        return ValueRange.of (1, InternationalFixedChronology.WEEKS_IN_MONTH);
+    }
+
+    @Override
+    InternationalFixedDate resolvePrevious (final int newYear, final int newMonth, final int dayOfMonth) {
+        return resolvePreviousValid (newYear, newMonth, dayOfMonth);
+    }
+
+    @Override
+    public ValueRange range (final TemporalField field) {
```

Comment:
You're going to need special values for day-of-week (and a few others), given that the leap-day and year-day aren't part of the week.  See how I handled this in the Discordian calendar (and some of the revisions Joda-Stephen made for related methods).